Assets/Scripts/CodeBlock.cs

```diff
@@ -33,6 +34,15 @@ public abstract class CodeBlock : MonoBehaviour {
             SetUpManipulationLogger();
             dragScript = gameObject.AddComponent<CloneOnDrag>(); // TODO: clean this up
             UpdateText();
+            startingRotation = transform.rotation;
+        }
+
+        private void Update(){
+            if(!isMenuBlock)
+            {
+                transform.rotation = startingRotation;
```

Comment:
do you need to do this every frame for some reason? Can this be moved to a function where it needs to happen? If it needs to happen when something is grabbed, there is an event for that as well